### ISSUE TITLE
bug: should use round-robin for kafka sink without PK specified

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave#13325]
> > > Not true, Kafka will use Round Robin for these messages and I don't think there is data skew.
> > 
> > 
> > Kafka would use Round Robin when key is unset. But our current implementation always sets key = {} (2 bytes). This is a minor implementation bug we need to fix.
> 
> OMG, that's a bug indeed.

Let's follow this?

https://github.com/apache/kafka/blob/f1e58a35d7aebbe72844faf3e5019d9aa7a85e4a/clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java#L28-L32

```
 * The default partitioning strategy:
 * <ul>
 * <li>If a partition is specified in the record, use it
 * <li>If no partition is specified but a key is present choose a partition based on a hash of the key
 * <li>If no partition or key is present choose the sticky partition that changes when the batch is full.
```

_Originally posted by @fuyufjh in https://github.com/risingwavelabs/risingwave/issues/9443#issuecomment-1803105297_

**Comment by fuyufjh** [comment]
Oops, already fixed.

**Comment by fuyufjh** [comment]
BTW, can you help to update the docs as well? The docs said the 'primary key' is optional  https://docs.risingwave.com/docs/dev/create-sink-kafka/ cc. @xiangjinwu

**Comment by xiangjinwu** [comment]
fixed by #12810

corresponding doc update: https://github.com/risingwavelabs/risingwave-docs/pull/1371#pullrequestreview-1681273562
